buildenv/Dockerfile

```diff
@@ -20,7 +20,7 @@
 # SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0-only WITH Classpath-exception-2.0 OR GPL-2.0-only WITH OpenJDK-assembly-exception-1.0
 #
 
-FROM ubuntu:22.04@sha256:6042500cf4b44023ea1894effe7890666b0c5c7871ed83a97c36c76ae560bb9b
+FROM public.ecr.aws/lts/ubuntu:22.04_stable@sha256:1582c29f34a48752e406f1a261fe9545fad895da3f6bb4be55bc82485557564e
```

Comment:
How does this improve security? Either way by specifying a SHA256 hash, we'll get the image we expect.

Also, why is the hash changing?

Reply:
It's not an attempt to improve security, it's just a change to use amazon to get the image instead of dockerhub. I hard coded the hash I got from pulling the amazon image. The latest hash at dockerhub is different from what we've been using, we should be regularly updating the image to the latest which we haven't been doing. The latest hash at dockerhub isn't the same as the one at amazon, possibly they are updated at different times and don't necessary have the same content.